torchrec/csrc/dynamic_embedding/details/id_transformer_variant.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <torchrec/csrc/dynamic_embedding/details/id_transformer_variant.h>
+#include <vector>
+
+namespace torchrec {
+
+IDTransformerVariant::IDTransformerVariant(
+    LXUStrategyVariant strategy,
+    int64_t num_embeddings,
+    const std::string& id_transformer_type)
+    : strategy_(std::move(strategy)),
+      var_(NaiveIDTransformer<uint32_t>(num_embeddings)) {
+  if (id_transformer_type != "naive") {
```

Comment:
This check should not be done here (should be somewhere in the upper levels). Thus you don't need to pass id_transformer_type at all.